src/test/ui/associated-const/issue-69201.rs

```diff
@@ -0,0 +1,23 @@
+// Tests that associated constants are checked whether they are used or not.
+//
+// revisions: used unused
+// compile-flags: -Copt-level=2 --emit link
+
+#![cfg_attr(unused, allow(dead_code))]
+#![deny(arithmetic_overflow)]
+
+pub trait Foo {
+    const N: i32;
+}
+
+struct S;
+
+impl Foo for S {
+    const N: i32 = 1 << 42;
```

Comment:
You called this `issue-69201`, but that issue contains a testcase and it is different from this one. Not sure what happened here. (The testcase from #69201 already exists in-tree, it is in `ui/lint/lint-exceeding-bitshifts.rs`, marked "FIXME".)

Reply:
Oops, yeah -- I completely missed that there was already a test case.